tests/components/bayesian/test_binary_sensor.py

```diff
@@ -293,3 +300,12 @@ def test_observed_entities(self):
 
         state = self.hass.states.get("binary_sensor.test_binary")
         assert ["sensor.test_monitored"] == state.attributes.get("observed_entities")
+
+        self.hass.states.set("sensor.test_monitored1", "on")
+        self.hass.block_till_done()
+
+        state = self.hass.states.get("binary_sensor.test_binary")
+        assert [
+            "sensor.test_monitored",
+            {"sensor.test_monitored1", "sensor.test_monitored"},
```

Comment:
You are not allowed to have any sets in state attributes. This breaks the frontend. You need to flatten this and remove duplicates and only calculate it once.

Reply:
Thanks for the feedback. I've now flattened it and removed the duplicates.